## Exercise 2.1

1. We’ve seen that n = 42 is legal. What about 42 = n?

In [1]:
42 = n

SyntaxError: cannot assign to literal (<ipython-input-1-f0c37ed646fc>, line 1)

ANS: We get a syntax error.

2. How about x = y = 1?

In [2]:
x = y = 1

ANS: First, 1 is assigned to y and the value of y is assigned to x later. So, we will not get any syntax error here.

3. In some languages every statement ends with a semi-colon, ;. What happens if you put a
semi-colon at the end of a Python statement?

In [3]:
print("Hello, World!");

Hello, World!


ANS: It executes normally.

4. What if you put a period at the end of a statement?

In [4]:
print("Hello, World!").

SyntaxError: invalid syntax (<ipython-input-4-995315a3f26e>, line 1)

ANS: We get a syntax error.

5. In math notation you can multiply x and y like this: xy. What happens if you try that in
Python?

In [5]:
xy

NameError: name 'xy' is not defined

ANS: The Python interpreter will think 'xy' as variable. Since, the value is not assigned, it will give out error.

## Exercise 2.2

1. The volume of a sphere with radius r is 4/3
πr^3
. What is the volume of a sphere with radius 5?

In [6]:
pi = 3.1415926535897932
r = 5
volume = (4/3) * pi * (r**3)
print(volume)

523.5987755982989


ANS: The volume of a sphere with radius 5 is 523.60

2. Suppose the cover price of a book is \\$24.95, but bookstores get a 40% discount. Shipping costs
$3 for the first copy and 75 cents for each additional copy. What is the total wholesale cost for
60 copies?


In [7]:
bookCost = 24.95
totalBooks = 60

cost = ((bookCost * 0.60) * totalBooks)
shippingCost = (3 + 0.75 * (totalBooks - 1))
totalCost = cost + shippingCost
print('Total Wholesale cost', totalCost)

Total Wholesale cost 945.4499999999999


ANS: The total wholesale cost for 60 copies is $945.45